lib/utils/cloud.ts

```diff
@@ -106,3 +106,105 @@ export const getDeviceAndMaybeAppFromUUID = _.memoize(
 	// Memoize the call based on UUID
 	(_sdk, deviceUUID) => deviceUUID,
 );
+
+/**
+ * Download balenaOS image for the specified `deviceType`.
+ * `OSVersion` may be one of:
+ *  - exact version number,
+ *  - valid semver range,
+ *  - `latest` (includes pre-releases),
+ *  - `default` (excludes pre-releases if at  least one stable version is available),
+ *  - `recommended` (excludes pre-releases, will fail if only pre-release versions are available),
+ *  - `menu` (will show the interactive menu )
+ * If not provided, OSVersion will be set to `default`
+ *
+ * @param deviceType
+ * @param outputPath
+ * @param OSVersion
+ */
+export async function downloadOSImage(
+	deviceType: string,
+	outputPath: string,
+	OSVersion?: string,
+) {
+	const unzip = await import('node-unzip-2');
+	const fs = await import('fs');
+	const manager = await import('balena-image-manager');
+
+	console.info(`Getting device operating system for ${deviceType}`);
+
+	if (!OSVersion) {
+		console.warn(`OS version is not specified, using the default version: \
+the newest stable (non-pre-release) version if available, or the newest version otherwise \
+(if all existing versions for the given device type are pre-release).`);
+	}
```

Comment:
I'll update this later with the new short version.

Reply:
Then I must not approve the PR yet, as bulldozer would merge it if I did... :-)